www/content/essays/no-build-step.md

```diff
@@ -0,0 +1,68 @@
++++
+title = "Why htmx Does Not Have a Build Step"
+date = 2023-08-19
+[taxonomies]
+author = ["Alexander Petros"]
+tag = ["posts"]
++++
+
+A recurring question from some htmx contributors is why htmx isn't written in TypeScript, or, for that matter, why htmx lacks any build step at all. The full htmx source is a single 3,500-line JavaScript file; if you want to contribute to htmx, you do so by modifying the `htmx.js` file, the same file that gets sent to browsers in production, give or take minification and compression.
+
+I do not speak for the htmx project, but I have made a few nontrivial contributions to it, and have been a vocal advocate for retaining this no-build setup every time the issue has arisen. From my perspective, here's why htmx does not have a build step.
+
+## Write Once, Run Forever
+The best reason to write a library plain JavaScript is that it lasts forever. This is arguably JavaScript's single most underrated feature. While I'm sure there are some corner cases, JavaScript from 1999 that ran in Netscape Navigator will run unaltered, alongside modern code, in Google Chrome downloaded yesterday. That is true for very few programming environments. It's certainly not true for Python, or Java, or C, which all have versioning mechanisms where opting for new language features will force you off of deprecated APIs.
+
+Of course, most people's experience with JavaScript is that is ages like than milk. Reopen a node repository after 3 months and you'll find that your project is mired in a flurry of security warnings, backwards-incompatible library "upgrades," and a frontend framework whose cultural peak was the exact moment you started the project and is now widely considered tech debt. Who's to blame for this situation is for someone else to decide, but, in any case, this entire problem class is eliminated if you depend on nothing but the JavaScript runtime.
```

Comment:
"is ages like than milk" --> "it ages faster than milk"

Reply:
changing to "ages like milk" (the "than" was a typo)